test/integration/functional_test.go

```diff
@@ -1782,6 +1783,80 @@ users:
 	}
 }
 
+func validateStartWithCorpProxy(ctx context.Context, t *testing.T, profile string) {
+	if !GithubActionRunner() {
+		t.Skip("Only run mitmproxy test on github actions")
+	}
+
+	defer PostMortemLogs(t, profile)
+
+	// Pull down the mitmproxy docker image
+	dockercmd := exec.CommandContext(ctx, "docker", "pull", "mitmproxy/mitmproxy")
+	_, err := Run(t, dockercmd)
+	if err != nil {
+		t.Fatalf("Failed to download mitmproxy docker image: %v", err)
+	}
+
+	certDir, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatalf("creating temp dir failed: %v", err)
+	}
+
+	// Start an interactive session (since mitmproxy requires it) asyncronously
+	// This will create the necessary .mitmproxy directory with its certs
+	mitmCmd := exec.CommandContext(ctx, "docker", "run", "-it", "--rm", "--name", "mitmproxy", "-v", certDir, ":/home/mitmproxy/.mitmproxy", "-p", "8080:8080", "mitmproxy/mitmproxy")
```

Comment:
this might cause docker rate limit. I recommend using non docker registry or install the binary on the github action maybe?

Reply:
yeah we can create our own docker image with the mitmproxy binary if necessary

Reply:
we're using a binary directly now, so this is no longer an issue